postgres/osmap.yaml

```diff
@@ -91,9 +91,46 @@ RedHat:
 {% endif %}
 
 Suse:
+  pkg_repo:
+    name: pgdg-sles-{{ release }}
+    humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
+    #Using sles-12 upstream repo for opensuse
+    baseurl: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch'
+    key_url: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch/repodata/repomd.xml.key'
+    gpgcheck: 1
+    gpgautoimport: True
+
+{% if repo.use_upstream_repo %}
+  {# Pillars needed are 'use_upstream_repo: True' and 'version: n.n'. #}
+  {# Avoid setting package names as pillars, as may corrupt postgres. #}
+  {% set lib_dir = '/var/lib/pgsql/' ~ repo.version ~ '/data' %}
+
+  pkg: postgresql{{ release }}-server
+  pkg_client: postgresql{{ release }}
+  conf_dir: {{ lib_dir }}
+   {% if release|int == 10 %}
+  service: postgresql-{{ release }}
```

Comment:
I've checked out `postgresql96-server-9.6.6-1PGDG.sles12.x86_64.rpm` package, and there the service name follows the same pattern. I think we should omit release version check above.